### PR TITLE
plugin: rewrite model train py file.

### PR DESCRIPTION
**Pull request by Txiaozhe**
The last pr for #105 :)

### CODE REVIEW

File: packages/plugins/model-train/bayesian-model-train/src/script.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+  row_data.forEach((data, i) => {
+    const word_cut: string[] = jieba.cut(data, boa.kwargs({
+      cut_all: false
+    }));
+    data_list.push(word_cut);
+    class_list.push(row_class[i]);
+  });
+      
+  // 划分训练集和测试集
```
Comment (yorkie) [inline packages/plugins/model-train/bayesian-model-train/src/script.ts:55]: Use English comments please :)

File: packages/plugins/model-train/bayesian-model-train/src/script.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  const data_class_list = zip(data_list, class_list);
+  random.shuffle(data_class_list);
+  const [ train_data_list, train_class_list ] = unzip(data_class_list);
+
+  interface AllWordsDict {
+    [key: string]: [ string, number ];
+  }
```
Comment (yorkie) [inline packages/plugins/model-train/bayesian-model-train/src/script.ts:62]: Could you please move the interface on the top level?

Comment (Txiaozhe) [reply packages/plugins/model-train/bayesian-model-train/src/script.ts:62]: Of cause~

File: packages/plugins/model-train/bayesian-model-train/src/script.ts (hunk 1 above)
Comment (Txiaozhe) [reply packages/plugins/model-train/bayesian-model-train/src/script.ts]: fixed.

**Comment by utkobe** [comment]
@Txiaozhe Please run the test pipeline `text-bayes-classification.json` in test/pipelines, it seems not working for me. To run it, just use
```
sh run_pipeline.sh text-bayes-classification
```

**Comment by yorkie** [comment]
> @Txiaozhe Please run the test pipeline `text-bayes-classification.json` in test/pipelines, it seems not working for me. To run it, just use
> 
> ```
> sh run_pipeline.sh text-bayes-classification
> ```

How about enabling `text-bayes-classification` in our PR workflow? That makes sense for:

- validating if pipeline workflow is working
- checking our Boa-based plugin is working
- and this model is small enough that the duration is acceptable to put it in our PR workflow.

> I'm not requesting to do above in this PR, we can try that in another PR.

**Comment by Txiaozhe** [comment]
> @Txiaozhe Please run the test pipeline `text-bayes-classification.json` in test/pipelines, it seems not working for me. To run it, just use
> 
> ```
> sh run_pipeline.sh text-bayes-classification
> ```

Yeah, running pipline is necessary~

**Comment by Txiaozhe** [comment]
> > @Txiaozhe Please run the test pipeline `text-bayes-classification.json` in test/pipelines, it seems not working for me. To run it, just use
> > ```
> > sh run_pipeline.sh text-bayes-classification
> > ```
> 
> How about enabling `text-bayes-classification` in our PR workflow? That makes sense for:
> 
> * validating if pipeline workflow is working
> * checking our Boa-based plugin is working
> * and this model is small enough that the duration is acceptable to put it in our PR workflow.
> 
> > I'm not requesting to do above in this PR, we can try that in another PR.

Running pipline in pr workflow is a good idea so that pipline can be run automatic.

**Comment by Txiaozhe** [comment]
There are some error when I install deps for pipcook and boa so thatI can not run pipline local, could I run the pipline in the pr workflow now ? @yorkie

**Comment by yorkie** [comment]
> There are some error when I install deps for pipcook and boa so thatI can not run pipline local, could I run the pipline in the pr workflow now ? @yorkie

You could rebase your PR when #125 gets merged.

**Comment by Txiaozhe** [comment]
> > There are some error when I install deps for pipcook and boa so thatI can not run pipline local, could I run the pipline in the pr workflow now ? @yorkie
> 
> You could rebase your PR when #125 gets merged.

Okay, thanks~

**Comment by Txiaozhe** [comment]
Can I remove this code:
```js
    const module = boa.import('script');
    const importlib = boa.import('importlib');
    importlib.reload(module);
```
it will throw error for boa.
@yorkie

**Comment by yorkie** [comment]
What errors do you get? I think we still need to track the reason in this case.

**Comment by Txiaozhe** [comment]
> What errors do you get? I think we still need to track the reason in this case.

'script' module not found,
because we have removed the script.py to script.ts?

**Comment by yorkie** [comment]
Ah, yea, this is because the module not found.

**Comment by Txiaozhe** [comment]
> Ah, yea, this is because the module not found.

en，I removed it.

**Comment by Txiaozhe** [comment]
The pipline completed local.



@utkobe @yorkie

**Comment by Txiaozhe** [comment]
Install deps failed because of the bad proxy :(

**Comment by yorkie** [comment]
It seems that this PR decreases the model accuracy to 0.3:

```
Dumping model to file cache /tmp/jieba.cache
Loading model cost 0.834 seconds.
Prefix dict has been built successfully.
current execution component: modelEvaluate
evaluate result:  { accuracy: 0.33377837116154874 }
```

**Comment by Txiaozhe** [comment]
same model accuracy result on master:

```shell
➜  pipcook git:(master) sh run_pipeline.sh text-bayes-classification
start execution: 
pipeline id: d5ff2630-85fd-11ea-9118-51382ce3c1a7
current execution component: dataCollect
downloading dataset ...
 ████████████████████████████████████████ 100% | ETA: 0s | 136168/136168
unzip and collecting data...
current execution component: dataAccess
current execution component: modelDefine
current execution component: modelTrain
Building prefix dict from the default dictionary ...
Loading model from cache /tmp/jieba.cache
Loading model cost 0.618 seconds.
Prefix dict has been built successfully.
current execution component: modelEvaluate
evaluate result:  { accuracy: 0.33377837116154874 }
Pipline is completed.
➜  pipcook git:(master) 

```

**Comment by Txiaozhe** [comment]
This pr not merged on master.

**Comment by yorkie** [comment]
See https://github.com/alibaba/pipcook/runs/614487185, it's 0.92 on our latest master build.

**Comment by yorkie** [comment]
@Txiaozhe May I ask you to rebase this PR based on our master and try again? I follow this and it works for me.

**Comment by Txiaozhe** [comment]
> @Txiaozhe May I ask you to rebase this PR based on our master and try again? I follow this and it works for me.

Of cause, but I get 0.92 after I rebuild it, so may be some problems in my pr, I am checking and fix it.

**Comment by Txiaozhe** [comment]
> @Txiaozhe May I ask you to rebase this PR based on our master and try again? I follow this and it works for me.

I rebase to latest merge but get 0.33 still :(

**Comment by utkobe** [comment]
I think the reason is that in there are some difference between the former python version and boa version. For example, the python will convert generator to string with set operator and this is not gonna happen in python. My suggestion is to try to check output of each boa function in bayesian-model-train and compare the output with former python version. @Txiaozhe

**Comment by Txiaozhe** [comment]
> I guess the reason is that in there are some difference between the former python version and boa version. For example, the python will convert generator to string with set operator and this is not gonna happen in node. My suggestion is to try to check output of each boa function in bayesian-model-train and compare the output with former python version. @Txiaozhe

Thanks, I will try it.

**Comment by Txiaozhe** [comment]
throw err when I run the pipline but I have not do any code changing:

```shell
pipeline id: ac555490-8615-11ea-b9c2-31615da1a2ab
current execution component: dataCollect
downloading dataset ...
 ████████████████████████████████████████ 100% | ETA: 0s | 136168/136168
unzip and collecting data...
current execution component: dataAccess
current execution component: modelDefine
current execution component: modelTrain
Building prefix dict from the default dictionary ...
Loading model from cache /tmp/jieba.cache
Loading model cost 0.614 seconds.
Prefix dict has been built successfully.
libc++abi.dylib: terminating with uncaught exception of type pybind11::error_already_set: SyntaxError: ('invalid syntax', ('<string>', 2, 42, '[1 if word in set() else 0 for word in 新款,满减,包邮,正品,简约,韩版,红包,创意,时尚,优惠,休闲,家用,品牌,学生,特价,工厂,厂家直销,长袖,儿童,优惠券,男女,套装,复古,家居,秋冬,定制,手机,全场,客厅,修身,男士,实木,进口,品质,官方,促销,外套,特产,宽松,现代,运动,企业,户外,女装,加厚,宝宝,原创,秋季,手工,个性,韩国,店铺,潮流,真皮,保湿,大码,全店,纯棉,男装,防水,卧室,透气,组合,家具,百搭,
```

the error happened on:
```js
export const TextFeatures = function(train_data_list: string[], feature_words: string[]) {
  function text_features(text: string, feature_words: string[]) {
    const text_words = set(text);
    return boa.eval(`[1 if word in ${text_words} else 0 for word in ${feature_words}]`);
  }

  return train_data_list.map((text: string) => {
    return text_features(text, feature_words);
  });
};
```

**Comment by yorkie** [comment]
You need tagged string template:

```js
boa.eval`[1 if word in ${text_words} else 0 for word in ${feature_words}]`;
```

**Comment by Txiaozhe** [comment]
I think this problem may not on this pr, because 0.33 is the model evaluation value, and this value is related to the model definition and model training, so I think should check the previous pr too.

**Comment by Txiaozhe** [comment]
jieba.cut returned unavailable data:

**Pull request merged**